### PR TITLE
Fix filename of exported dataset to support millisecond precision

### PR DESCRIPTION
**Pull request by glombiewski**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

Filenames for different timesteps of exported datasets only had seconds in their filename. If a tilmestep had only differences in milliseconds, this caused errors.

### CODE REVIEW

**Comment by glombiewski** [comment]
[Dateiname bei Erstellung von Dataset über Tasks hat nur Sekundengenauigkeit](https://trello.com/c/s7FSgirZ/1165-dateiname-bei-erstellung-von-dataset-%C3%BCber-tasks-hat-nur-sekundengenauigkeit)

**Comment by ChristianBeilschmidt** [comment]
[Deploybare Version des EBV-Analyzers erstellen](https://trello.com/c/rCZOfj4w/1169-deploybare-version-des-ebv-analyzers-erstellen)

File: operators/src/util/raster_stream_to_geotiff.rs [hunk 1]
```diff
@@ -285,7 +285,7 @@ impl<P: Pixel + GdalType> GdalDatasetHolder<P> {
             full_path: placeholder_path,
             placeholder: placeholder.to_string(),
             time_placeholder: GdalSourceTimePlaceholder {
-                format: DateTimeParseFormat::custom("%Y-%m-%dT%H-%M-%S".to_string()),
+                format: DateTimeParseFormat::custom("%Y-%m-%dT%H-%M-%S-%3f".to_string()),
```
Comment (ChristianBeilschmidt) [inline operators/src/util/raster_stream_to_geotiff.rs:288]: Could you add a test, where it is checked (in a temp folder) that files with specific (expected) names exist?

It is to some degree concerning that the file names changed but there was no test to adapt.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: operators/src/util/raster_stream_to_geotiff.rs [hunk 2]
```diff
@@ -1337,4 +1342,118 @@ mod tests {
...
+                    as_cog: true,
+                    compression_num_threads: GdalCompressionNumThreads::AllCpus,
+                    force_big_tiff: false,
+                },
+                None,
+                Box::pin(futures::future::pending()),
+                tiling_specification,
+            )
+            .await
+            .expect("");
```
[the remark this answers is not in the document]
Comment (glombiewski) [reply operators/src/util/raster_stream_to_geotiff.rs]: 9feac082b5e85103cc04712c756cb7b77cc1b9db

File: operators/src/util/raster_stream_to_geotiff.rs (hunk 1 above)
Comment (glombiewski) [reply operators/src/util/raster_stream_to_geotiff.rs:288]: ef3749e2769b7bf09ce3e11dbbeb3e597c4e97e5

**Review by ChristianBeilschmidt — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
bors merge

**Pull request merged**